README.md

```diff
@@ -97,11 +99,12 @@ is valid Python code for using [the integrator interface](https://diffeq.sciml.a
 
 ```py
 from diffeqpy import de
+import numpy as np
 
-def f(u,p,t):
-    return -u
+def f(du,u,p,t):
+    du[0] = -u[0]
 
-u0 = 0.5
+u0 = np.array([0.5])
```

Comment:
Is there a reason why scalars are no longer supported?

Reply:
With PythonCall, a python `float` is passed to Julia as an `Py` with a `pytype` of float. This object works in many places where one would expect a float, but not all. For example, it's not a `Number` and is not `iterable` (iteration falls back to python's iteration, and `float` is not iterable in python) so the check `any(isnan, x)` will error if we pass in a python float.

It's possible to use scalars, but they must be converted when they are passed into a differential equations problem (not too hard) and also whenever they are returned from a function that computes derivatives (somewhat harder).

Similarly, any use of non-mutating derivative functions requires the passed object to be converted from a python object to a Julia object, and the PythonCall semantics aren't quite what I'd like for that. For example, if a python function returns `x = [1,2,3]` we can call `pyconvert(Any, x)` which will convert it to a `PyList{Any}`, and the `eltype` of `Any` will cause errors down the line.

These conversion issues can be overcome, but I worry they would be brittle and could more than double the size of the diffeqpy codebase.

Reply:
I don't think it would be too bad to add a PythonCall extension where at the DEProblem constructors and such we can wrap the `f` to "do the right thing" and convert the u0/p types.

Reply:
Okay, I'll give it a shot.